src/edu/macalester/comp127/FarmArt/Menu.java

```diff
@@ -0,0 +1,177 @@
+package edu.macalester.comp127.FarmArt;
+
+import java.util.List;
+import edu.macalester.graphics.CanvasWindow;
+import edu.macalester.graphics.Image;
+import edu.macalester.graphics.Rectangle;
+import edu.macalester.graphics.ui.Button;
+ 
+/**
+ * Represent a menu on the canvas with zoom in/out button, drag button, element button, and background button
+ * To allow users modifier their farm
+ */
+public class Menu {
+
+   private Brush currentBrush;
+   private Rectangle buttonIndicator;
+   private Boolean dragging = false;
+ 
+   /**
+    * A menu constructor with mouse drag and mouse down lanbdas
+    * the menu contains the buttons and controls the dragging and brush applications
+    * @param canvas
+    * @param tileManager
+    * @param buttonIndicator To indicate user of their current chosen button
+    */
+    public Menu(CanvasWindow canvas, 
+                TileManager tileManager, 
+                Rectangle buttonIndicator, 
+                List<BackgroundType> backgroundsList, 
+                List<ElementType> elementsList) {
+        this.buttonIndicator = buttonIndicator;
+        addZoomButtons(canvas, tileManager);
+        addDragButton(canvas);
+        orderBackgroundButtons(canvas, backgroundsList);
+        orderElementButtons(canvas, elementsList);
+
+
+        canvas.onMouseDown(event -> {
+            if (!dragging) {
+                Tile tile = tileManager.findTileAt(event.getPosition());
+                if (tile != null && currentBrush != null) {
+                    currentBrush.apply(tile);
+                }
+            }
+        });
```

Comment:
I appreciate that your good class structure and method names lead to nice readable sections of code like this.